src/main/java/de/tum/in/www1/artemis/web/rest/ExamUserResource.java

```diff
@@ -75,9 +81,19 @@ public ResponseEntity<ExamUser> updateExamUser(@RequestPart ExamUserDTO examUser
                 .orElseThrow(() -> new EntityNotFoundException("Exam user with login: \"" + examUserDTO.login() + "\" does not exist"));
 
         if (signatureFile != null) {
-            String responsePath = fileService.handleSaveFile(signatureFile, true, false).toString();
-            examUser.setSigningImagePath(responsePath);
+            String oldPathString = examUser.getSigningImagePath();
+            Path basePath = FilePathService.getExamUserSignatureFilePath();
+            Path savePath = fileService.saveFile(signatureFile, basePath, false);
+            examUser.setSigningImagePath(FilePathService.publicPathForActualPathOrThrow(savePath, examUser.getId()).toString());
+
+            if (oldPathString != null) {
+                // Only delete old file if saving the new one succeeded
+                Path oldPath = FilePathService.actualPathForPublicPath(URI.create(examUser.getSigningImagePath()));
```

Comment:
I think, this immediately deletes the file at the new path

```suggestion
                Path oldPath = FilePathService.actualPathForPublicPath(URI.create(oldPathString));
```